src/main/java/fi/aalto/cs/apluscourses/dal/APlusTokenAuthentication.java

```diff
@@ -0,0 +1,39 @@
+package fi.aalto.cs.apluscourses.dal;
+
+import org.apache.http.HttpRequest;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+public class APlusTokenAuthentication extends TokenAuthentication {
+
+  public static final String AUTHORIZATION_HEADER = "Authorization";
+
+  public static final String APLUS_USER = "A+_user";
```

Comment:
Wasn't this supposed to be the API URL of the course or was that somewhere else?

Reply:
That is another field. For one service, there could be multiple credentials (that is, different users) stored. As discussed with @superseacat today, we could, in future, store actual user name to this field (once it's fetched from the API) allowing multiple users from the same IDE (submitting the same solutions? ;) ).

Reply:
Ah right, that makes sense